ext/dom/document.c

```diff
@@ -1633,8 +1634,8 @@ static void _dom_document_schema_validate(INTERNAL_FUNCTION_PARAMETERS, int type
 	switch (type) {
 	case DOM_LOAD_FILE:
 		if (CHECK_NULL_PATH(source, source_len)) {
-			php_error_docref(NULL, E_WARNING, "Invalid Schema file source");
-			RETURN_FALSE;
+			zend_argument_value_error(1, "must not contain any null bytes");
```

Comment:
In all three cases, you can also convert the `source_len == 0` case to throw `must not be empty`.

Reply:
Yes I was already planning to target these as well. Since this PR is specifically about handling of NULL bytes I did not want to add them now.

I will add them in a separate commit, that will not hurt.